bsip-0040.md

```diff
@@ -101,7 +111,10 @@ The assert that needs to pass is now `current_cumsum + incoming value <= max_cum
 #### `limit_monthly`
 Statefull assert, only `int` type arguments. Analogue to `limit`, but `interval_began` is initially set to `month(valid_from)` and set to `month(now)` on update, additionally the time assert is `month(now) >= interval_began + interval_in_months` (include logic for month overflow when year changes).
 
-### Simple Example
+#### `contains_only`
+Stateless assert, only for dictionary type objects like `options`. The `argument` of the operation is a dictionary type objects and is only allowed to contain the attributes in the data list.
```

Comment:
Most of "options" are not "dictionary type objects" if I understood your definition correctly. In the case of `asset_update_operation`, all attributes/fields inside the "options" need to be explicitly specified.

There ARE some dictionary type objects, for example `extensions` field of `account_create_operation`, `account_update_operation` and `call_order_update_operation`, all fields in the `extension` are `optional` so can be omitted.

Reply:
For both cases it is dictionary type right?

Diference is that in one case everything needs to be filled (and if unchanged with the onchain value) and the other case allows to only set the desired change.

I have removed `contains only` and added `attribute_assert` which has reduced logic to what it can do. The logic is not supposed to know which arguments are reflected on-chain an thus cant check if a value is unchanged. See `attribute_assert`.